[WIP] Improves RH branding

a2batic opened the pull request:
Login Screen
![screenshot from 2018-10-10 11-01-22](https://user-images.githubusercontent.com/12200504/46717986-7592b400-cc87-11e8-9bc1-8e452e42da64.png)

Navigation Bar
![screenshot from 2018-10-10 11-05-44](https://user-images.githubusercontent.com/12200504/46718001-817e7600-cc87-11e8-9e30-f192882d06f6.png)

About modal
![screenshot from 2018-10-10 11-06-03](https://user-images.githubusercontent.com/12200504/46718017-893e1a80-cc87-11e8-817c-9338365d6ac7.png)


Fixes: #22
Signed-off-by: Kanika Murarka <kmurarka@redhat.com>


<!--
Thank you for opening a pull request!  Here are some tips on creating
a well formatted contribution.

Please give your pull request a title like "[component]: [short description]"

This is the format for commit messages:

"""
[component]: [short description]

[A longer multiline description]

Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
Signed-off-by: [Your Name] <[your email]>
"""

The Signed-off-by line is important, and it is your certification that
your contributions satisfy the Developers Certificate or Origin.  For
more detail, see SubmittingPatches.rst.

The component is the short name of a major daemon or subsystem,
something like "mon", "osd", "mds", "rbd, "rgw", etc. For ceph-mgr modules,
give the component as "mgr/<module name>" rather than a path into pybind.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview.  More information for contributors is available here:
https://raw.githubusercontent.com/ceph/ceph/master/SubmittingPatches.rst

-->

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.html:
```diff
@@ -1,73 +1,60 @@
+            <div class="login-logo">
+              <img src="/assets/branding/Logo_RH_RGB_Reverse.png" class="redhat_logo" alt= "Redhat">
+            </div>
+            <footer class="login-pf-page-footer">
```
Based on [Patternfly login page](https://www.patternfly.org/pattern-library/application-framework/login-page/#design) as well, the footer links seem to be placed together with the main form.

a2batic replied:
@epuertat It is based on the branding [presentation](https://docs.google.com/presentation/d/1uLTmaNtdn0hgWSk5N9i5c4amXlpLqMVsOsHTfxx02wY/edit?usp=sharing) prepared by @julienlim

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.html:
```diff
@@ -1,73 +1,60 @@
+            </div>
+            <footer class="login-pf-page-footer">
+              <ul class="login-pf-page-footer-links list-unstyled">
+                <li><a class="login-pf-page-footer-link" href="#">Terms of Use</a></li>
```
@julienlim any idea on how to create these contents? Or is it better to remove them? At least the legally binding contents (`Terms of Use` and `Privacy Policy` should be provided by the legal team).

Should we understand `Help` as a general help (RHCS docs) or login page specific?

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss:
```diff
@@ -1,31 +1,21 @@
-    margin-bottom: 30px;
-  }
+.branding-border {
+  width: calc(100% + 60px);
```
What are these two first one for (`width: 100% + 60px` and `margin-left: -30px`)?

a2batic replied:
@epuertat RCUE branding doesn't seem to provide a top red border for login view. Since login div is child view of app component, I had to come up with this work-around. I added a div and styled it as the border. The width and margin are styling the same div.

epuertat replied:
I've only seen the red stripe at login screen in that screenshot provided by @julienlim, but [RCUE does not show that element](https://brand.redhat.com/applications/product-ux/). In Patternfly it exists (blueish) but is part of the login screen [background image](https://www.patternfly.org/components/patternfly/dist/img/bg-login.jpg). [Other screenshots of Red Hat products' login screens](https://www.google.com/search?q=red+hat+login&tbm=isch) don't show that red stripe either. My point is to just keep the branding as standard and straightforward as possible, given the upstream is not patternfly based. If that red stripe is not mandatory, I'd avoid it, as long as adding that on our own will require us to check that with different screen sizes and devices.

julienlim replied:
@a2batic @epuertat Regarding the red strip at the top of the login screen, I just spoke with Kyle Baker, who tells me the red stripe is optional now.  As it's no longer mandatory, let's leave it off to make the branding effort easier.

Just curious, is there any particular reason why we have extra hint words in the input text boxes?  Specifically, do we really need to be verbose and say "Enter your username" (vs. "Username") and "Enter your password" (vs. "Password")?  Is this due to downstream?

Otherwise, everything looks good!

epuertat replied:
@julienlim Thanks for the clarification! @a2batic has an exceptional eye for the detail (I completely overlooked that strip/stripe). The thing is that we still have the upstream branding recommendations by Michael pending to do, and that's gonna change completely the login screen layout, so perhaps that's a good chance for improving this.

a2batic replied:
@julienlim @epuertat ack, I will remove the top red border and "Enter your" phrase.

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss:
```diff
@@ -1,31 +1,21 @@
-  }
+.login-pf-page-footer{
+  width:30%;
+  position:fixed;
```
Maybe this is the setting that we should tune to put the footer together with the main form?

a2batic replied:
@epuertat It is based on the branding [presentation](https://docs.google.com/presentation/d/1uLTmaNtdn0hgWSk5N9i5c4amXlpLqMVsOsHTfxx02wY/edit?usp=sharing) prepared by @julienlim

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/about/about.component.html:
```diff
@@ -1,26 +1,35 @@
+  <h1>Red Hat Ceph Storage Dashboard</h1>
+  <div class="product-versions-pf">
+    <ul class="list-unstyled">
+      <li><strong>Version</strong> 3.2</li>
```
May we use [this solution](https://www.jpkeisala.com/setting-global-constants-angular-2-enviroment/) or [this other one](https://stackoverflow.com/a/47653328) to use global constant strings and group them into a single file (constants.ts)?

```ts
// constants.ts
PRODUCT_NAME = "Red Hat Ceph Storage Dashboard";
PRODUCT_VERSION = "3.2"
PRODUCT_TRADEMARK = `
Licensed under Creative Commons Attribution Share
Alike 3.0 (CC-BY-SA-3.0).
`
```
```html
...
  <h1>{{ PRODUCT_NAME }}</h1>
  <div class="product-versions-pf">
    <ul class="list-unstyled">
      <li><strong>Version</strong> {{ PRODUCT_VERSION }}</li>
...
   <div class="trademark-pf">
     {{ PRODUCT_TRADEMARK }}
   </div>
 ```

a2batic replied:
ack

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/about/about.component.html:
```diff
@@ -1,26 +1,35 @@
+      <li><strong>Version</strong> 3.2</li>
+      <li><strong>Ceph Manager</strong> {{ hostName }}</li>
+      <li><strong>User</strong> {{user}}</li>
+      <li><strong>User Role</strong> admin</li>
```
Can't this be obtained from the localStorage or other service? In case it cannot, I think it's better not to put it, as non-admin users will see admin role here.

a2batic replied:
ack, it can be obtained from users api

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/about/about.component.ts:
```diff
@@ -28,6 +35,11 @@ export class AboutComponent implements OnInit, OnDestroy {
+    this.browserName = detect().name;
+    this.browserVersion = detect().version;
+    this.browserOS = detect().os;
+    this.hostName = "Host";
```
You can easily obtain the hostname from the getHealth() (health/ endpoint) JSON:

This is the ceph-mgr service endpoint:
`mgr_map.active_address = "172.17.0.2:6800/11514"`

And this is the dashboard specific one:
`mgr_map.services.dashboard = "https://ceph-dev:11000/"`

In any case, that's not a plain hostname but an service (protocol + hostname + port), so we should change the name of the property as well.

a2batic replied:
@epuertat ack, maybe host address will be more accurate

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.html:
```diff
@@ -1,11 +1,10 @@
        href="#">
-      <img src="assets/Ceph_Logo_Standard_RGB_White_120411_fa.png"
-           alt="Ceph">
+      <img src="/assets/branding/RHCSD-Masthead-Logo.svg" width=100%  alt="Red Hat Ceph Storage">
```
If we use contants.ts `{{ PRODUCT_NAME }}`

a2batic replied:
ack

a2batic commented on src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.html:
```diff
@@ -1,73 +1,60 @@
+<div class="login-pf-page">
+  <div class="branding-border"></div>
+  <div class="container-fluid">
+    <div class="row">
```
ack

a2batic commented:
Screenshots after addressing comments:-

Login Screen: 
![screenshot from 2018-10-15 17-20-56](https://user-images.githubusercontent.com/12200504/46949447-1aa1f800-d09f-11e8-9025-0328d32764d8.png)

Navigation:-
![screenshot from 2018-10-15 17-21-16](https://user-images.githubusercontent.com/12200504/46949472-27bee700-d09f-11e8-86bf-a09a2aebe354.png)

About Modal:
![screenshot from 2018-10-15 17-21-31](https://user-images.githubusercontent.com/12200504/46949512-3c02e400-d09f-11e8-9bbb-0e8c668030ac.png)

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/branding.service.ts:
```diff
@@ -0,0 +1,37 @@
+
+  constructor(private dashboardService: DashboardService, private userService: UserService) {}
+
+  setContants() {
```
Typo: missing 's' in ConStants. Nonetheless, I think it's a bit misleading to have a method called 'setConstants' (that can be called multiple times), given that constants are usually pre-set and never set multiple times. Additionally, some of these are not even constants (`data.mgr_map.services.dashboard` can change if a Ceph-Mgr failover happens).

IMHO, this service is a mixture of 2 different things:
- **String constants**: product name, company, version, license string, copyright notice, etc.
- **Environment details**: current Ceph-mgr hostname, user name, role, browser, ...

So my suggestion would be:
- Have a separate constants file for real constants, I [already dropped a few example links](https://stackoverflow.com/questions/34986922/define-global-constants-in-angular-2).
- Have a service (if you want) for getting environment details/info, but I wouldn't call it branding, as it has nothing to do with branding. 

I think both pieces could perfectly fit upstream (the environment info, and the global constants).

a2batic replied:
ack

julienlim replied:
@a2batic @epuertat 

I meant to ask about the About Modal.

The Host Address seems to show a URL.  It used to show the Ceph Manager host(s).  Is the Host Address showing the Ceph Manager API endpoint or other API end point?  If so, the label may need to change to reflect that, but let's determine what the intent was first.

Does the User and User Role now dynamically update?

epuertat replied:
I think there are two possible sources for the IP address/hostname in the mgr_map (which is the source used for this): one is the current active Ceph-mgr hostname, and the other is the back-end service endpoint (a URL schema: http[s]://<ip>:<port>).

a2batic replied:
@julienlim @epuertat, 'User' is updating as expected, 'User Role' API endpoint is only accessible to users with admin permissions, there are two possibilities: either we remove 'User Role' or show it for admins only.

a2batic replied:
ack

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/login.service.ts:
```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
```
A few things: `login.service` is rather misleading, as this has little to do with the `auth/login` part of the code, so that could confuse people.

On the other hand, this service actually takes lots of information pieces from other services and adds little value (no processing or transformation, just the info grouped), so my suggestion would be to put all this stuff into the about box component itself, and remove the login.service. I don't think this service could be easily reusable by someone else.

epuertat commented on src/pybind/mgr/dashboard/frontend/src/app/core/navigation/about/about.component.ts:
```diff
@@ -1,36 +1,33 @@
-import { Subscription } from 'rxjs';
 
-import { SummaryService } from '../../../shared/services/summary.service';
+import { AboutConstants } from '../../../shared/enum/about.enum';
```
See my comment below on this.
